etc/dbus-serialbattery/daly.py

```diff
@@ -246,13 +246,16 @@ def read_cells_volts(self, ser):
             buffer[1] = self.command_address[0]  # Always serial 40 or 80
             buffer[2] = self.command_cell_volts[0]
 
-            maxFrame = int(self.cell_count / 3) + 1
+            if (int(self.cell_count) % 3) == 0:
+                maxFrame = (int(self.cell_count / 3))
+            else:
+                maxFrame = (int(self.cell_count / 3) + 1)
             lenFixed = (
-                maxFrame * 12
-            )  # 0xA5, 0x01, 0x95, 0x08 + 1 byte frame + 6 byte data + 1byte reserved
+                maxFrame * 13
+            )  # 0xA5, 0x01, 0x95, 0x08 + 1 byte frame + 6 byte data + 1byte reserved + chksum
 
             cells_volts_data = read_serialport_data(
-                ser, buffer, self.LENGTH_POS, self.LENGTH_CHECK, lenFixed
+                ser, buffer, self.LENGTH_POS, 0, lenFixed
```

Comment:
the normal call is not working on these multi-sentence transmissions. so just receive the full length and calculate the checksums in the cellvolts function later